### PR TITLE
Added sample to author and consume nuget package for xml transform using msbuild targets

### PR DESCRIPTION
**Pull request by jainaashish**
Added sample to author and consume nuget package for xml transform using msbuild targets for `PackageReference` world. This PR adds two things:
- Author, which tells how to author a NuGet package to transform `web.config` and `appconfig.config` files using `.transform` file through msbuild targets.  
- Consumer, which tells how to consume a package with `PackageReference` which transform xml files like `web.config` or `appconfig.config` through mabuild targets.

@rrelyea

### CODE REVIEW

File: XDTransformExample/authoring/pack.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$NuGetExe = Join-Path $PSScriptRoot '.nuget\nuget.exe'
+
+# Download NuGet.exe if missing
+if (-not (Test-Path $NuGetExe)) {
+    Write-Host 'Downloading nuget.exe'
+    New-Item  -ItemType directory  (Join-Path $PSScriptRoot '.nuget') | Out-Null
+    wget https://dist.nuget.org/win-x86-commandline/v3.3.0/nuget.exe -OutFile $NuGetExe
```
Comment (emgarten) [inline XDTransformExample/authoring/pack.ps1]: Use 4.4.1?

File: XDTransformExample/authoring/build/XMLTransformExample.targets [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+ 
+  <Target Name="CollectConfigFilesTransforms" BeforeTargets="CollectWebConfigsToTransform">
```
Comment (emgarten) [inline XDTransformExample/authoring/build/XMLTransformExample.targets]: Add a comment telling the author to rename this, if multiple packages have the same target only 1 will win

@nkolev92 added an `<Error>` to help solve this

Comment (nkolev92) [reply XDTransformExample/authoring/build/XMLTransformExample.targets]: Pointed to the sample offline. 
https://github.com/NuGet/Samples/blob/master/NuGet.Samples.Interop/NuGet.Samples.Interop.targets#L17

Putting a random placeholder which would make the target not interpretable also works.

**Comment by ryno1234** [comment]
I cannot seem to get this to work for our project.  A few questions before I spend any more time debugging the issue:

* Is this meant to work when the target project is an SDK-style project using the new csproj format?
* Is the transform supposed to occur during installation of the Nuget package, during build of the consuming project or during publish of the consuming project?

**Comment by karann-msft** [comment]
@ryno1234 xdt transforms don't work with packageReference based projects and by extension, SDK-style ones https://docs.microsoft.com/en-us/nuget/reference/migrate-packages-config-to-package-reference#xdt-transforms-are-not-applied-when-the-package-is-installed-after-the-upgrade

**Comment by ryno1234** [comment]
@karann-msft , Understood... however I thought the whole purpose of this sample was to demonstrate how to address that deficiency.

See https://github.com/NuGet/Home/issues/5954

**Comment by karann-msft** [comment]
Ah yes. You are right. I haven't tried this sample lately. Can you file an issue on the home repo with steps to repro the issue, expected, and actual behavior. Thanks!

**Pull request merged**